### PR TITLE
Time series

### PR DESCRIPTION
**Pull request by JonathanChavezTamales**
# Summary

This PR integrates 3 new charts and 1 test:

- **Percentage of HTTPS websites of all websites**
- **Percentage of HTTPS requests of all websites**
- **Percentage of resources with HSTS header**

All of them are Line Charts to represent a time series, the purpose of them is to show the growth over time of these variables.

I also added the first test to a backend endpoint.

- [x] Add test `/top-websites-with-mixed-content`

## How these tests work?

There's a folder inside the server code called `__mocks__/@google-cloud` which has a mock class for BigQuery and the query method is mocked so that it does not call BigQuery but fetches results from a local .json file. If there's anything wrong, please let me know.

Inside that mock there's a function called `simulateQueryResult` which takes the query passed to BigQuery.query and returns a 'mockedResult' which is in queries.json, this way we don't have to repeat the work and just call the results from that file and have a single source of truth.

There's also another folder called `__tests__` where API endpoints are tested with *supertest* using the mock previously mentioned.

### Notes

@ernestognw is working in reorganizing the layout. This PR won't affect the UI because he branched from this branch.

Big thanks to @ernestognw who taught us how to test the backend using Node.js!

### CODE REVIEW

**Comment by ernestognw** [comment]
@googlebot I consent

File: server/__tests__/api.test.js [hunk 1]
```diff
@@ -1,10 +1,30 @@
 import request from 'supertest';
 import app from '../app';
```
Comment (ernestognw) [inline server/__tests__/api.test.js:2]: nit: Alphabetize

File: src/views/mixed-content/index.js [hunk 2]
```diff
@@ -1,3 +1,8 @@
+import HTTPSPercentagePages
+  from './https-percentage-pages';
+import HTTPSPercentageRequests
+  from './https-percentage-requests';
+import HSTSPercentageRequests from './hsts-percentage-requests';
```
Comment (ernestognw) [inline src/views/mixed-content/index.js:5]: nit: Alphabetize

File: server/__tests__/api.test.js [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
+
+/**
+ * Test suite for mixed-content endpoint (api/mixed-content)
+ */
+describe('mixed-content', () => {
+  it('Check /top-websites-with-mixed-content', async () => {
+    const res = await request(app).get('/api/mixed-content/'+
+    'top-websites-with-mixed-content');
+
+    expect(res.body.description).toBe('Top websites with most mixed content'+
```
Comment (ernestognw) [inline server/__tests__/api.test.js:25]: There is a missing space here that is making the tests to fail.

Just fix it and we're ready to go

Comment (JonathanChavezTamales) [reply server/__tests__/api.test.js:25]: Done!

File: src/views/mixed-content/https-percentage-requests/index.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+import {Typography, Snackbar} from '@material-ui/core';
+
+const HTTPSPercentageRequests = () => {
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState([]);
+  const [snackOpen, setSnackOpen] = useState(false);
+
+  useEffect(() => {
+    api
+        .get('/api/mixed-content/https-percentage-requests')
```
Comment (livvielin) [inline src/views/mixed-content/https-percentage-requests/index.js:18]: It seems like you have similar functionality in other files. Is it possible to put this into a common utils? Fine to do this in a separate PR.

Comment (JonathanChavezTamales) [reply src/views/mixed-content/https-percentage-requests/index.js:18]: Sure, I converted this into a GitHub issue.

File: src/views/mixed-content/hsts-percentage-requests/elements.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+
+const ChartContainer = styled.div`
+  height: 300px;
+  display: flex;
+  justify-content: center;
+  align-items: center
```
[the remark this answers is not in the document]
Comment (JonathanChavezTamales) [reply src/views/mixed-content/hsts-percentage-requests/elements.js:7]: Done!

[the remark this answers is not in the document]
Comment (JonathanChavezTamales) [reply src/views/mixed-content/hsts-percentage-requests/elements.js:7]: What I'm looking is that every component folder has the same file with the same contents, is that supposed to be that way @ernestognw ? or can we refactor that?

[the remark this answers is not in the document]
Comment (ernestognw) [reply src/views/mixed-content/hsts-percentage-requests/elements.js]: You're correct Jonathan. At this point, every container is almost the same, but that doesn't mean they're equal. Once layout is refactored, every ChartContainer will vary, so even if its styles are the same, their context and application are different.

Later we can create a folder with templates with repeated logic, but it is too early to do it since we don't really know what the patterns are yet.
